editor/src/components/assets.ts

```diff
@@ -42,20 +42,54 @@ export function getAllProjectAssetFiles(
   return allProjectAssets
 }
 
-function getSHA1Checksum(contents: string): string {
+function getSHA1Checksum(contents: string | Buffer): string {
   return new sha1().update(contents).digest('hex')
 }
 
-export function getProjectContentsChecksums(tree: ProjectContentTreeRoot): GithubChecksums {
+export function inferGitBlobChecksum(buffer: Buffer): string {
+  // This function returns the same SHA1 checksum string that git would return for the same contents.
+  // Given the contents in the buffer variable, the final checksum is calculated by hashing
+  // a string built as "<prefix><contents>". The prefix looks like "blob <contents_length_in_bytes><null_character>".
+  // Ref: https://git-scm.com/book/en/v2/Git-Internals-Git-Objects
+  const prefix = Buffer.from(`blob ${buffer.byteLength}\0`)
+  const wrapped = Buffer.concat([prefix, buffer])
+  return getSHA1Checksum(wrapped)
+}
+
+export function getProjectContentsChecksums(
+  tree: ProjectContentTreeRoot,
+  assetChecksums: GithubChecksums,
+): GithubChecksums {
   const contents = treeToContents(tree)
 
   const checksums: GithubChecksums = {}
   Object.keys(contents).forEach((filename) => {
     const file = contents[filename]
-    if (isTextFile(file)) {
-      checksums[filename] = getSHA1Checksum(file.fileContents.code)
-    } else if (isAssetFile(file) && file.base64 != undefined) {
-      checksums[filename] = getSHA1Checksum(file.base64)
+    if (file == null) {
+      return
+    }
+
+    switch (file.type) {
+      case 'TEXT_FILE':
+        checksums[filename] = getSHA1Checksum(file.fileContents.code)
+        break
+      case 'ASSET_FILE':
+        if (file.base64 != undefined) {
+          checksums[filename] = getSHA1Checksum(file.base64)
+        }
+        break
+      case 'IMAGE_FILE':
+        if (file.gitBlobSha) {
```

Comment:
```suggestion
        if (file.gitBlobSha != null) {
```
The same logic from here should also apply to `ASSET_FiLE` values as well.

Reply:
:+1: https://github.com/concrete-utopia/utopia/pull/2933/commits/f1a2dd824dec2a94cd48f6940380be822a48e9d4